specification/sections/06-tracking.md

```diff
@@ -0,0 +1,40 @@
+---
+title: Tracking
+description: Specification defining tracking API
+toc_max_heading_level: 4
+---
+
+# 6. Tracking
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+## Overview
```

Comment:
There are two other important use cases which could be worth calling out

- flag usage analysis (for detecting obsolete flags)
- controlled rollout: making sure a flag change has no unexpected impact on metrics (which typically includes more technical than business metrics)

Not sure how much these would actually affect how the spec is defined, but as it stands the scope seems a bit too narrow

Reply:
I believe those use cases fall under the observability initiative https://github.com/open-telemetry/community/pull/2042, but it would not hurt to check for overlap with tracking.

Reply:
I'm totally on board with the "controlled rollout" use case but please explain a bit more about "flag usage analysis".

Reply:
Flag usage analysis: tracking which flags are no longer being evaluated, so they can be retired.

But I guess that's probably out of scope for this, because it can probably be implemented entirely within the SDK and/or provider, without needing the app dev to explicitly call anything

Reply:
> I believe those use cases fall under the observability initiative [open-telemetry/community#2042](https://github.com/open-telemetry/community/pull/2042), but it would not hurt to check for overlap with tracking.

Our o11y initiative would only help to support controlled rollout if both the provider and the flag consumer are using otel, AND the flag consumer is willing to ship otel traces to the flag provider. I believe that a lot of providers will want an alternative to that - i.e. manually instrumenting the metrics that they want to observe during rollout. 

Also, the metrics which are tracked during a controlled rollout can vary from very technical (and thus likely to be measured via otel) to biz-oriented (and thus much more likely to be tracked via a segment-style tracking API). We should at least call out what's in scope, what's handled elsewhere, and what we're putting on for now

Reply:
Oh and also any OF/otel integration is unlikely to help at all for client-side flags, at least for today. AFAIK it's still pretty rare to instrument client-side code with otel (although hopeful that'll change over time)

Reply:
> Flag usage analysis: tracking which flags are no longer being evaluated, so they can be retired.
> 
> But I guess that's probably out of scope for this, because it can probably be implemented entirely within the SDK and/or provider, without needing the app dev to explicitly call anything

Yes, and AFAIK it can be implemented already (though it's not standardized in any way) without the addition of this tracking API. What you're describing would be based on telemetry around impressions... a very important topic, but orthogonal to this initiative.